keras/engine/saving.py

```diff
@@ -501,7 +507,46 @@ def preprocess_weights_for_loading(layer, weights,
                                                           weights[num_weights_per_layer:],
                                                           original_keras_version,
                                                           original_backend)
-        weights = forward_weights + backward_weights
+        return forward_weights + backward_weights
+
+    def convert_nested_model(weights):
+        """
+        Converts layers nested in `Model` or `Sequential` by `preprocess_weights_for_loading()`.
+        :param weights: list of of numpy arrays
+        :return: converted list of of numpy arrays
+        """
+        new_weights = []
+        # trainable weights
+        for sublayer in layer.layers:
+            num_weights = len(sublayer.trainable_weights)
+            if num_weights > 0:
+                new_weights.extend(preprocess_weights_for_loading(
+                    layer=sublayer,
+                    weights=weights[:num_weights],
+                    original_keras_version=original_keras_version,
+                    original_backend=original_backend))
+                weights = weights[num_weights:]
+
+        # non-trainable weights
+        for sublayer in layer.layers:
+            num_weights = len([l for l in sublayer.weights
```

Comment:
Weights are either trainable or not. Why is this necessary?

Reply:
Well, I just moved code from one place to another. I'm not sure why somebody wrote it like this. IMHO refactoring that code is for another PR.

Reply:
Ok, SG